src/diff/index.js

```diff
@@ -110,6 +110,11 @@ export function diff(dom, parentDom, newVNode, oldVNode, context, isSvg, excessD
 					break outer;
 				}
 
+				if (isProvider && c.props.value!==newVNode.props.value) {
```

Comment:
Is this basically the same as changing the original `componentDidUpdate` on Provider to `componentWillUpdate`?

Reply:
Hmm, yes that's a good point. Not fully sure about this approach yet personally but shared it already since it seems to solve an issue that is open at this point.